src/runtime/platform/NativeCodePage.cs

```diff
@@ -246,9 +246,9 @@ public void SetReadExec(IntPtr mappedMemory, int numBytes)
         /// </summary>
         public static void InitializePlatformData()
         {
-            // FIXME: arch, non-windows
-            Machine = Runtime.Is32Bit ? MachineType.i386 : MachineType.x86_64;
-            OperatingSystem = Runtime.IsWindows ? OperatingSystemType.Windows : OperatingSystemType.Linux;
+            MachineName = SystemInfo.GetArchitecture();
+            Machine = SystemInfo.GetMachineType();
+            OperatingSystem = SystemInfo.GetSystemType();
```

Comment:
I think we should remove all of these. @filmor thoughts?

Reply:
I agree.